pkg/collector/runner/running_checks_tracker.go

```diff
@@ -0,0 +1,75 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package runner
+
+import (
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/collector/check"
+)
+
+// RunningChecksTracker is an object that keeps a thread-safe track of
+// all the running checks
+type RunningChecksTracker struct {
```

Comment:
Do we need all those methods/Objects to be part of the public API of the `runner` package ?

Reply:
Made a bunch of the public methods/objects private